service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java

```diff
@@ -142,97 +138,93 @@ public OrderTypeAndDocument buildOrderTypeAndDocument(OrderTypeAndDocument typeA
             .build();
     }
 
-    public Map<String, Object> getOrderTemplateData(CaseData caseData,
+    public DocmosisGeneratedOrder getOrderTemplateData(CaseData caseData,
```

Comment:
Do we want to split this out into its own service like the others, extending `DocmosisTemplateDataGeneration`? Also would we want a separate service for each one or do we want to keep it all clustered together in one? It would probably be easier to test if they are separated but it would mean that there is some duplication unless there was a common super class between it and `DocmosisTemplateDataGeneration`. Maybe something like
- `DocmosisTemplateDataGeneration`
  - `GeneratedOrderTemplateData`
    - `BlankOrder....`
    - `CareOrder...`
    - `SupervisionOrder...`
    - `EPO...`

Reply:
Thanks for pointing this out, I will make it extend DocmosisTemaplteDataGeneration.